### ISSUE TITLE
Missile Command Bug: UserMissile line sometimes does not terminate

### ISSUE THREAD

**Issue by Matthew-Herman** [Matthew-Herman/personal-website#1]
Sometimes UserMissile doesn't terminate from arrFiredMissiles once reaching xTarget and yTarget positions. 

This could be an error related to JS forcing floating points, and if so then this problem could be fixed by checking for inequality instead of equality.

**Comment by Matthew-Herman** [comment]
Fixed for when cursor is about 100px above bottom of screen by adding a buffer area to the collision detection. Problem still exists when cursor is less than 100px above bottom of screen. Could be problem in the calculation.

One possible fix is instead of checking for collision between UserMissile.xPos and UserMissile.yPos and UserMissile.xTarget and UserMissile.yTarget, only check if xPos and yPos is outside a virtual circle centered at UserMissile.xStart and UserMissile.yStart.

**Comment by Matthew-Herman** [comment]
Solved the bug. 

The problem was that I set UserMissile.ySpeed to be proportional to the distance from the start position, and then set UserMissile.xSpeed based upon that. When ySpeed was very small, xSpeed was so large that the object would skip over the buffer area, in which collision between the UserMissile and the target point could be detected. 

I solved the issue by changing the movement algorithm to have a constant UserMissile.speed variable, and then set xSpeed and ySpeed using trigonometry. Specifically as follows:
```javascript
this.direction = Math.atan((this.xTarget-this.xStart)/(this.yTarget-this.yStart));
this.ySpeed = -this.speed*Math.cos(this.direction);
this.xSpeed = -this.speed*Math.sin(this.direction);
```
This has permanently fixed the issue, and the same change could be made on EnemyMissile but it's not necessary to add the additional computations I think.
